File: src/app/(tabs)/test/-components/1.js
class MainActivity extends ComponentActivity {
    onCreate(savedInstanceState) {
        super.onCreate(savedInstanceState);

        enableEdgeToEdge();

        setContent({}, () => {
            MyApplicationTheme({}, () => {
                Scaffold({ modifier: Modifier.fillMaxSize() }, (innerPadding) => {
                    Counter({
                        name: "Android",
                        modifier: Modifier.padding(innerPadding),
                    });
                });
            });
        });
    }
}

/**
 * @Composable
 */
function Counter({ name, modifier = Modifier }) {
    const count = remember(0);

    return Column(
        {
            modifier: [modifier, Modifier.alignCenter()],
        },
        () => {
            Text({
                text: `Hello ${name}!`,
            });
            Text({
                text: String(count),
            });
            Spacer((modifier = Modifier.height()));
            Row(
                {
                    horizontalArrangement: Arrangement.Center,
                    verticalAlignment: Alignment.CenterVertically,
                },
                () => {
                    Button({
                        text: "-",
                        onClick: () => count - 1,
                    });
                    Button({
                        text: "+",
                        onClick: () => count + 1,
                    });
                },
            );
        },
    );
}

File: src/app/(tabs)/test/-components/1.kt
package com.example.myapplication

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.myapplication.ui.theme.MyApplicationTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            MyApplicationTheme {
                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
                    Counter(
                        name = "Android",
                        modifier = Modifier.padding(innerPadding)
                    )
                }
            }
        }
    }
}

@Composable
fun Counter(name: String, modifier: Modifier = Modifier) {
    var count by remember { mutableIntStateOf(0) }

    Column(
        modifier = modifier.fillMaxSize(), // Applied once to the root container
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "Hello $name!" // Removed duplicate modifier usage here
        )

        Text(
            text = "$count",
            style = MaterialTheme.typography.displayLarge
        )

        Spacer(modifier = Modifier.height(24.dp))

        Row(
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Button(onClick = { count-- }) {
                Text(text = "-", fontSize = 24.sp)
            }

            Spacer(modifier = Modifier.width(16.dp))

            Button(onClick = { count++ }) {
                Text(text = "+", fontSize = 24.sp)
            }
        }
    }
}